fossunited/ticketing/doctype/foss_event_ticket/test_foss_event_ticket.py

```diff
@@ -1,74 +1,24 @@
-# Copyright (c) 2024, Frappe x FOSSUnited and Contributors
-# See license.txt
-
-from datetime import datetime, timedelta
-
 import frappe
 from faker import Faker
 from frappe.tests import IntegrationTestCase
 
 from fossunited.api.checkins import checkin_attendee
-from fossunited.doctype_ids import CHAPTER, CONFERENCE, EVENT, EVENT_TICKET, USER_PROFILE
+from fossunited.doctype_ids import CHAPTER, EVENT, EVENT_TICKET
+from fossunited.tests.utils import generate_test_chapter, generate_test_event
 
 
 class TestFOSSEventTicket(IntegrationTestCase):
     def setUp(self):
-        fake = Faker()
-
-        chapter_member = frappe.db.get_value(USER_PROFILE, {"user": "test1@example.com"}, ["name"])
-        self.member_email = "test1@example.com"
-        self.member_profile = chapter_member
-
-        if not frappe.db.exists("Role", "Chapter Team Member"):
-            frappe.get_doc({"doctype": "Role", "role_name": "Chapter Team Member"}).insert(
-                ignore_permissions=True
-            )
-        if not frappe.db.exists("Role", "Chapter Lead"):
-            frappe.get_doc({"doctype": "Role", "role_name": "Chapter Lead"}).insert(
-                ignore_permissions=True
-            )
-
-        chapter = frappe.get_doc(
-            {
-                "doctype": CHAPTER,
-                "chapter_type": CONFERENCE,
-                "chapter_name": fake.name(),
-                "email": fake.email(),
-                "about_chapter": fake.text(),
-                "chapter_members": [{"chapter_member": chapter_member}],
-            }
+        self.lead_email = "test1@example.com"
+        self.chapter = generate_test_chapter(lead_email=self.lead_email)
+        self.event = generate_test_event(
+            chapter=self.chapter, is_paid_event=True, tickets_status="Live"
         )
-        chapter.insert(ignore_permissions=True)
-
-        event = frappe.get_doc(
-            {
-                "doctype": EVENT,
-                "chapter": chapter.name,
-                "event_name": fake.name(),
-                "event_permalink": fake.slug().replace("-", "_"),
-                "status": "Live",
-                "event_type": CONFERENCE,
-                "event_start_date": datetime.today(),
-                "event_end_date": datetime.today() + timedelta(1),
-                "event_description": fake.text(),
-                "is_paid_event": 1,
-                "tickets_status": "Live",
-                "tiers": [
-                    {
-                        "enabled": 1,
-                        "title": "General",
-                        "price": 100,
-                    }
-                ],
-            }
-        )
-        event.insert(ignore_permissions=True)
-
-        self.event = event
 
     def tearDown(self):
         frappe.set_user("Administrator")
         frappe.delete_doc(EVENT, self.event.name, force=True)
+        frappe.delete_doc(CHAPTER, self.chapter.name, force=True)
```

Comment:
we're deleting `event` and then `chapter` here but earlier in this PR, we were doing `chapter` and then `event`. which is the right order?

`chapter` first and then `event` makes sense to me because `event` depends on `chapter` if i remember correctly.

again, good to use a consistent convention across the codebase - you never know which testcase will be copied to create a new one, worsening this problem in the future.

Reply:
doesn't really matter, since its all being force deleted